services/placement/service/placementservice.go

```diff
@@ -147,30 +147,30 @@ func (ps placementService) AddReplica() (services.ServicePlacement, error) {
 
 func (ps placementService) AddInstance(
 	candidates []services.PlacementInstance,
-) (services.ServicePlacement, error) {
+) (services.ServicePlacement, services.PlacementInstance, error) {
 	p, v, err := ps.ss.Placement(ps.service)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	var addingInstance services.PlacementInstance
 	if addingInstance, err = ps.findAddingInstance(p, candidates, ps.opts); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if p, err = ps.algo.AddInstance(p, addingInstance); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if err := placement.Validate(p); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if ps.opts.Dryrun() {
 		ps.logger.Info("this is a dryrun, the operation is not persisted")
-		return p, err
+		return p, addingInstance, err
```

Comment:
this would return you the addingInstance with empty shards, you should probably return p.Instance(addingInstance.ID()) to get the added instance with shard assignment